### PR TITLE
xapi-stdext-unix: remove proxy()

### PR DESCRIPTION
**Pull request by lindig**
The proxy function has no clients and uses select() which limits the
number of FDs. Remove it.

Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

### CODE REVIEW

**Review by psafont — APPROVED** [review]
xcp-idl uses this function, but has its own copy, on `xcp-idl/lib/posix_channel.ml`

vncproxy used to use the function as well, but got archived.

**Comment by lindig** [comment]
I have added a commit that removes select() using epoll(). I've tested this manually. It relies on the Polly package.

File: lib/xapi-stdext-unix/unixext.ml [hunk 1]
```diff
@@ -446,52 +446,74 @@ let really_write_string fd string =
...
-    now := Unix.gettimeofday()
-  done;
-  if !bytes_written = total_bytes_to_write then () else (* we ran out of time *) raise Timeout
+let defer f g = Fun.protect ~finally:f g
+
+let write_with_timeout write fd length bytes timeout =
+  let deadline = Unix.gettimeofday () +. timeout in
+  let written = ref 0 in
+  let epoll = Polly.create () in
+  defer (fun () -> Polly.close epoll) @@ fun () ->
```
Comment (psafont) [inline lib/xapi-stdext-unix/unixext.ml:455]: I don't understand the interaction and the `Polly.close`, it looks like it's closed on every iteration, is that safe to do?

Comment (lindig) [reply lib/xapi-stdext-unix/unixext.ml:455]: The defer is outside the `loop` - it is closed once

Comment (psafont) [reply lib/xapi-stdext-unix/unixext.ml:455]: That's right, I somehow confused `process` with the outer function

**Review by psafont — APPROVED** [review]

**Comment by psafont** [comment]
Could docstrings be added to both functions in the mli?

**Comment by lindig** [comment]
This might not be safe to remove; there is a client in xapi:

```
ocaml/vncproxy/vncproxy.ml:             Xapi_stdext_unix.Unixext.proxy s fd)))
```

**Comment by lindig** [comment]
Closing this as we first need to rewrite clients.

**Pull request closed without merging**